.github/workflows/ci.yml

```diff
@@ -24,36 +27,49 @@ jobs:
     steps:
     - uses: actions/checkout@v2
 
-    - name: build Docker image
+    - name: Build Docker image
       run: docker build -t wlan-cloud-ucentralgw:${{ github.sha }} .
 
-    - name: log into Docker registry
-      uses: docker/login-action@v1
-      with:
-        registry: ${{ env.DOCKER_REGISTRY_URL }}
-        username: ${{ env.DOCKER_REGISTRY_USERNAME }}
-        password: ${{ secrets.DOCKER_REGISTRY_PASSWORD }}
-
-    - name: push Docker image
+    - name: Tag Docker image
       run: |
         TAGS="${{ github.sha }}"
 
-        if [ ${GITHUB_REF} == "refs/heads/master" ]
+        if [[ ${GITHUB_REF} == "refs/heads/"* ]]
         then
-          TAGS="$TAGS ${GITHUB_REF#refs/heads/}"
-        else # PR build
-          CURRENT_TAG=$(echo ${GITHUB_HEAD_REF#refs/heads/} | tr '/' '-')
+          CURRENT_TAG=$(echo ${GITHUB_REF#refs/heads/} | tr '/' '-')
           TAGS="$TAGS $CURRENT_TAG"
+        else
+          if [[ ${GITHUB_REF} == "refs/tags/"* ]]
+          then
+            CURRENT_TAG=$(echo ${GITHUB_REF#refs/tags/} | tr '/' '-')
+            TAGS="$TAGS $CURRENT_TAG"
+          else # PR build
+            CURRENT_TAG=$(echo ${GITHUB_HEAD_REF#refs/heads/} | tr '/' '-')
+            TAGS="$TAGS $CURRENT_TAG"
+          fi
         fi
 
-        echo "Pushing tags $TAGS"
+        echo "Result tags: $TAGS"
 
         for tag in $TAGS; do
           docker tag wlan-cloud-ucentralgw:${{ github.sha }} ${{ env.DOCKER_REGISTRY_URL }}/ucentralgw:$tag
-          docker push ${{ env.DOCKER_REGISTRY_URL }}/ucentralgw:$tag
         done
 
+    - name: Log into Docker registry
+      if: startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master'
```

Comment:
Mmh this condition will also prevent (non-release branch) PR build pushes, right?

Reply:
Well that's a good question. Probably we do want it as we probably want to have PR build pushes as we would like to know if docker-compose deployment would work with all that
I'll add PRs in it too

Reply:
@brennerm , done, one more review round please?